include/ttmlir/Dialect/TTNN/IR/TTNNOps.td

```diff
@@ -146,6 +146,13 @@ def TTNN_NegOp : TTNN_ElementwiseUnaryOp<"neg"> {
     }];
 }
 
+def TTNN_LogicalNotOp: TTNN_ElementwiseUnaryOp<"logical_not"> {
```

Comment:
How do you know if an op is DPS op in TTNN lib?

Reply:
DPS ops in its API signature have optional output tensor:
const std::optional<Tensor>& optional_output_tensor = std::nullopt

Take a look for example in [unary_composite.hpp](https://github.com/tenstorrent/tt-metal/blob/3a9dfbf0e920f6daa842db744be731b09071a065/ttnn/cpp/ttnn/operations/eltwise/unary/unary_composite.hpp#L56C1-L63C3) which isn't DPS op. There is no optional_output_tensor:
```
template <UnaryCompositeOpType unary_comp_op_type>
struct ExecuteUnaryCompositeOp {
    static Tensor invoke(
        const Tensor& input_tensor, const std::optional<MemoryConfig>& memory_config = std::nullopt) {
        auto output_memory_config = memory_config.value_or(input_tensor.memory_config());
        return OpHandler<unary_comp_op_type>::handle(input_tensor, output_memory_config);
    }
};
```

If you take a look at the [unary.hpp](https://github.com/tenstorrent/tt-metal/blob/3a9dfbf0e920f6daa842db744be731b09071a065/ttnn/cpp/ttnn/operations/eltwise/unary/unary.hpp#L30C1-L36C77), this is DPS op, hence there is an optional_output_tensor:
```
template <UnaryOpType... unary_op_types>
struct ExecuteUnary {
    static Tensor invoke(
        uint8_t queue_id,
        const Tensor& input_tensor,
        const std::optional<MemoryConfig>& memory_config = std::nullopt,
        const std::optional<Tensor>& optional_output_tensor = std::nullopt);
```

Reply:
Hey @sdjordjevicTT, thanks for jumping in while I was out of office.

@mmanzoorTT "destination passing style" idiom was new to me up until a few months ago, here's a link with a nice explanation in case you want to read more about it: https://www.tweag.io/blog/2024-03-07-dps-haskell/#what-is-destination-passing-style-programming

Reply:
Please correct me if I am wrong. 
* `logical_and` and `logical_or` are defined in `binary.cpp` as `BinaryOperation`; which is DPS op.
* `logical_xor` is defined in `binary_composite.cpp` as `ExecuteBinaryCompositeOps`; which is non DPS op.
* `logical_not` is defined in `unary.cpp` as `ExecuteUnary` (which is DPS op); however in-place version is defined as composite op making it non-DPS.

So I should only add logical_xor as non-DPS and rest of the ops are DPS ops?

Reply:
thanks for confirmation. I'll update my PR :)

Reply:
I am covering `logical_and`, `logical_or`, and `logical_not` in this PR. I am working on `logical_xor` and it requires a bit more testing. I'll add support for `logical_xor` in separate PR.